core/src/test/java/apoc/export/ExportCoreSecurityTest.java

```diff
@@ -0,0 +1,189 @@
+package apoc.export;
+
+import apoc.ApocConfig;
+import apoc.export.csv.ExportCSV;
+import apoc.export.cypher.ExportCypher;
+import apoc.export.graphml.ExportGraphML;
+import apoc.export.json.ExportJson;
+import apoc.util.TestUtil;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.neo4j.configuration.GraphDatabaseSettings;
+import org.neo4j.graphdb.Result;
+import org.neo4j.test.rule.DbmsRule;
+import org.neo4j.test.rule.ImpermanentDbmsRule;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * @author mh
+ * @since 22.05.16
+ */
+@RunWith(Parameterized.class)
+public class ExportCoreSecurityTest {
+
+    private static File directory = new File("target/import");
+
+    static {
+        directory.mkdirs();
+    }
+
+    @ClassRule
+    public static DbmsRule db = new ImpermanentDbmsRule()
+            .withSetting(GraphDatabaseSettings.load_csv_file_url_root, directory.toPath().toAbsolutePath());
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        TestUtil.registerProcedure(db, ExportCSV.class, ExportJson.class, ExportGraphML.class, ExportCypher.class);
+    }
+
+    private final String exportMethod;
+
+    public ExportCoreSecurityTest(String exportMethod) {
+        this.exportMethod = exportMethod;
+    }
+
+
+    @Parameterized.Parameters
+    public static Collection<String> data() {
+        return Arrays.asList("csv", "json", "graphml", "cypher");
+    }
+
+    @Test
+    public void testIllegalFSAccessExportQuery() {
+        final String message = "apoc.export." + exportMethod + ".query should throw an exception";
+        try {
+            db.executeTransactionally("CALL apoc.export." + exportMethod + ".query(\"RETURN 'hello' as key\", './hello', {})", Map.of(),
+                    Result::resultAsString);
+            fail(message);
+        } catch (Exception e) {
+            assertExportError(e);
+        }
+        try {
+            db.executeTransactionally("CALL apoc.export." + exportMethod + ".query(\"RETURN 'hello' as key\", './hello', {stream:true})", Map.of(),
+                    Result::resultAsString);
+            fail(message);
+        } catch (Exception e) {
+            assertExportError(e);
```

Comment:
has something changed in this PR that makes the export fail? initial (internal) issue report stated that this export would (wrongly) succeed and we don't see anything in `checkWriteAllowed` that would prevent the export when `stream` is `true`

Reply:
This case should be allowed only if `apoc.export.file.enabled=true` but in those test cases we only test `apoc.export.file.enabled=false`